### PR TITLE
Refactoring and deadlock solution of `wazuhDBWrapper`

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|#23301|

## Objective
This PR aims to overhaul the `socketDBWrapper` class by implementing a blocking approach to socket operations and converting the class to use a singleton pattern. This refactor enhances the efficiency and manageability of socket interactions throughout our application. Additionally, we have refactored the unit tests to align with these changes and ensure comprehensive coverage and reliability.

## Description
- The existing implementation of the `socketDBWrapper` class was identified as a deadlock, particularly in scenarios with high concurrency and rapid socket operations.
- To address these issues, we have refactored the `socketDBWrapper` to a blocking model, which simplifies the management of socket operations and ensures that each operation is completed before proceeding. This change is expected to improve the stability and performance of network communications.
- Moreover, we have converted the `socketDBWrapper` into a singleton to prevent multiple instances from being created. This ensures that socket operations are centralized and managed consistently throughout the application's lifecycle.
- The unit tests for `socketDBWrapper` were also overhauled to reflect these changes. New tests were added to specifically assess the blocking behavior and the singleton instance management, ensuring that the class functions correctly under various scenarios and that thread safety and singleton properties are maintained.

### Static Analysis
- The refactored code underwent thorough static analysis to ensure that it adheres to best coding practices and standards. Special attention was paid to thread safety and synchronization mechanisms, given the transition to a blocking approach and singleton usage.

#### Unit Testing
- Specific tests were designed to ensure that the singleton instance is properly created and accessed and that no additional instances are instantiated throughout the application's operation.

#### Integration Testing
N/A

------------------------

### CODE REVIEW

File: src/shared_modules/utils/socketWrapper.hpp [hunk 1]
```diff
@@ -679,6 +679,7 @@ class Socket final : public T
             // Send the data.
             while (bufferSize != amountSent)
             {
+                std::cerr<<"Sending data to socket"<<std::endl;
```
Comment (Dwordcito) [inline src/shared_modules/utils/socketWrapper.hpp:682]: remove this

Comment (GabrielEValenzuela) [reply src/shared_modules/utils/socketWrapper.hpp]: Done. Thx!

File: src/shared_modules/utils/tests/socketDBWrapper_test.cpp [hunk 2]
```diff
@@ -14,19 +14,17 @@
...
 
 TEST_F(SocketDBWrapperTest, EmptyTest)
 {
     m_query = "SELECT * FROM test_table;";
-    m_responses = std::vector<std::string> {""};
+    m_responses = std::vector<std::string> {" "};
 
     nlohmann::json output;
-    SocketDBWrapper socketDBWrapper(TEST_SOCKET);
     // The exception captured here is the timeout
```
Comment (sebasfalcone) [inline src/shared_modules/utils/tests/socketDBWrapper_test.cpp:26]: Update comment, we no longer have timeouts

File: src/wazuh_modules/vulnerability_scanner/tests/unit/osDataCache_test.cpp [hunk 3]
```diff
@@ -10,6 +10,10 @@
  */
 
 #include "osDataCache_test.hpp"
+#include "MockSocketDBWrapper.hpp"
+#include "TrampolineSocketDBWrapper.hpp"
+
+auto constexpr EMPTY_RESPONSE = "";
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/tests/unit/osDataCache_test.cpp:16]: Unused variable
```suggestion
```

File: src/wazuh_modules/vulnerability_scanner/tests/component/vulnerabilityScannerFacade_test.cpp [hunk 4]
```diff
@@ -44,7 +46,7 @@ TEST_F(VulnerabilityScannerFacadeTest, TestSingletonOfTheVulnerabilityScannerFac
 /*
  * @brief Test start method of the VulnerabilityScannerFacade class.
  */
-TEST_F(VulnerabilityScannerFacadeTest, TestStartMethod)
+TEST_F(VulnerabilityScannerFacadeTest, TestStartToEndMethod)
 {
     // TODO: Remove GTEST_SKIP and add EXPECTS once the implementation of the 'Indexer Connector' module is completed
     GTEST_SKIP();
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/tests/component/vulnerabilityScannerFacade_test.cpp:52]: This test is disabled due to this, should we remove this line?

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/008/expected_001.out
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/008/expected_001.out:1]: Remove white-spaces

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/008/expected_002.out
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/008/expected_002.out:1]: Remove white-spaces

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/008/expected_003.out
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/008/expected_003.out:1]: Remove white-spaces

File: src/shared_modules/utils/tests/socketDBWrapper_test.cpp [hunk 5]
```diff
@@ -14,19 +14,17 @@
...
 // temp header
-#include "wazuhDBQueryBuilder.hpp"
 #include <chrono>
 #include <thread>
 
 TEST_F(SocketDBWrapperTest, EmptyTest)
 {
     m_query = "SELECT * FROM test_table;";
-    m_responses = std::vector<std::string> {""};
+    m_responses = std::vector<std::string> {" "};
```
Comment (sebasfalcone) [inline src/shared_modules/utils/tests/socketDBWrapper_test.cpp:23]: Why is there a space here? Aren't we testing an empty response?

**Review by sebasfalcone — COMMENTED** [review]
Nice work @GabrielEValenzuela!

File: src/wazuh_modules/vulnerability_scanner/tests/component/vulnerabilityScannerFacade_test.cpp (hunk 4 above)
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/component/vulnerabilityScannerFacade_test.cpp:52]: We should keep it

File: src/shared_modules/utils/tests/socketDBWrapper_test.cpp (hunk 5 above)
Comment (GabrielEValenzuela) [reply src/shared_modules/utils/tests/socketDBWrapper_test.cpp:23]: For the size of bytes to send :)

File: src/wazuh_modules/vulnerability_scanner/tests/unit/osDataCache_test.cpp (hunk 3 above)
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/unit/osDataCache_test.cpp:16]: Done :)

File: src/shared_modules/utils/tests/socketDBWrapper_test.cpp (hunk 2 above)
Comment (GabrielEValenzuela) [reply src/shared_modules/utils/tests/socketDBWrapper_test.cpp:26]: Done :)

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**